src/solver/infeasible-problem-analysis/include/antares/solver/infeasible-problem-analysis/watched-constraints.h

```diff
@@ -0,0 +1,87 @@
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Antares::Optimization
+{
+class WatchedConstraint
+{
+public:
+    void setConstraintName(std::string constraintName);
+    std::string regexId();
+    virtual std::shared_ptr<WatchedConstraint> clone() const = 0;
+    virtual std::string infeasisibity() = 0;
+    virtual std::string infeasisibityCause() = 0;
+
+protected:
+    std::string name_;
+    std::string regexId_;
+    std::vector<std::string> splitName_;
+};
+
```

Comment:
i think it's not necessary to have all these derived classes!
the only differrences beaing the regex and the infeasibility message

Reply:
If you suggest we make one class with 2 member functions, each of them being a switch or a series of if statement, I'm sorry to disagree. 
This is what we used to have, and what I want to avoid : it's contrary to good design rules.
Indeed, it violates SRP and fosters big classes/functions.

Reply:
and the later one can be deducefrom the constraint name 'after the split

Reply:
in fact std::string timeStep(std::vector<std::string> splitName) can also return the timstep type (hour, day or weekly)

Reply:
std::string **WatchedConstraint**::infeasisibity()
{
    return shortName(splitName_) + "' at " timeStepType_ + timeStep(splitName_);
}

Reply:
and infeasisibityCause seems to be constant for one type ;

Reply:
Bodies of **infeasisibity()** and **infeasisibityCause()** can be quite different from a class to another, how would you handle with the messages they return ?

Reply:
see my last comment for infeasisibity() method 
and the infeasibility cause can be (- const std::string member data) obtained from the constructor

Reply:
ok, so @a-zakir suggests that we change messages of **infeasibility** and **infeasibility cause**, that is standardize them, in order to have one constraint class instead of one per constraint.
@flomnes : what do you think ?